lib/Sema/CSDiag.cpp

```diff
@@ -3668,9 +3668,12 @@ static bool tryDiagnoseNonEscapingParameterToEscaping(Expr *expr, Type srcType,
     return false;
 
   // Must be from non-escaping function to escaping function
-  auto srcFT = srcType->getAs<AnyFunctionType>();
-  auto destFT = dstType->getAs<AnyFunctionType>();
-  if (!srcFT || !destFT || !srcFT->isNoEscape() || destFT->isNoEscape())
+  auto srcFT =
```

Comment:
Actually, this part seems wrong too: we can't assign from `@escaping (() -> Void)?` to `() -> Void` for multiple reasons. (Although we can from `(() -> Void)!`.)